app/code/Magento/Checkout/Model/GuestPaymentInformationManagement.php

```diff
@@ -88,12 +102,17 @@ public function savePaymentInformationAndPlaceOrder(
         try {
             $orderId = $this->cartManagement->placeOrder($cartId);
         } catch (\Magento\Framework\Exception\LocalizedException $e) {
+            /** @var $quoteIdMask QuoteIdMask */
+            $quoteIdMask = $this->quoteIdMaskFactory->create()->load($cartId, 'masked_id');
+            /** @var \Magento\Quote\Model\Quote $quote */
+            $quote = $this->cartRepository->getActive($quoteIdMask->getQuoteId());
+            $this->checkoutHelper->sendPaymentFailedEmail($quote, __($e->getMessage()));
```

Comment:
Not only payment methods can throw `LocalizedException`, your implementation will send email even if something wrong not only with a payment transaction.